### ISSUE TITLE
[gRPC] generated gRPCTargets contain hardcoded location instead of ${_IMPORT_PREFIX}

### ISSUE THREAD

**Issue by a-andre** [Microsoft/vcpkg#1810]
Since the qRPC update from 1.4.1 to 1.6.0 CMake cannot find imported gRPC libraries.

Lines in `gRPCTargets-*.cmake` look like
```cmake
list(APPEND _IMPORT_CHECK_FILES_FOR_gRPC::gpr "C:/vcpkg/packages/grpc_x64-windows/debug/lib/gpr.lib" )
```
while they should look like
```cmake
list(APPEND _IMPORT_CHECK_FILES_FOR_gRPC::gpr "${_IMPORT_PREFIX}/debug/lib/gpr.lib" )
```

I'm running vcpkg on Windows 10.

qRPC was build with `.\vcpkg install grpc:x64-windows`

**Comment by gwihlidal** [comment]
I'm hitting this problem as well (gRPC 1.6.x with CMake on Windows 10). It would be great to have a fix applied.

**Comment by KindDragon** [comment]
> Since the qRPC update from 1.4.1 to 1.6.0 CMake cannot find imported gRPC libraries.

Why CMake can't find libraries? Full path is correct

**Comment by a-andre** [comment]
The built libraries are distributed so their locations are not fixed.

**Comment by KindDragon** [comment]
Ah, so it works locally

**Comment by a-andre** [comment]
Problem seems to be that [`gRPC_INSTALL*DIR`](https://github.com/grpc/grpc/blob/master/CMakeLists.txt#L33) are absolute paths. Check the description of `DESTINATION` in https://cmake.org/cmake/help/v3.9/command/install.html#introduction.

Removing `${CMAKE_INSTALL_PREFIX}/` from `gRPC_INSTALL_{BIN,LIB,INCLUDE,CMAKE}DIR` could solve the problem.

**Comment by ras0219-msft** [comment]
TIL that CMake's variable types actually matter! Apparently, setting a cache variable to `PATH` type makes it absolute with respect to the current source dir.

I think I have a fix; I expect to check it in soon.

**Comment by ras0219-msft** [comment]
Thanks everyone for working on figuring this out! It should be fixed in f527812e.
